helpers/lib/resourceHints.js

```diff
@@ -0,0 +1,69 @@
+const utils = require("handlebars-utils");
+
+const resourceHintsLimit = 50;
+
+const defaultResourceHintState = 'preload';
+
+const resourceHintStates = [defaultResourceHintState, 'preconnect', 'prefetch'];
+
+const resourceHintFontType = 'font';
+const resourceHintStyleType = 'style';
+const resourceHintScriptType = 'script';
+const resourceHintAllowedTypes = [resourceHintStyleType, resourceHintFontType, resourceHintScriptType];
+
+/**
+ * @param {string} path - The uri to the resource.
+ * @param {string} state - 'preload' or 'preconnect'
+ * @param {string} type - any of [style, font, script]
+ */
+function addResourceHint(globals, path, state, type) {
+
+    if (typeof path !== 'string') {
+        throw new Error('resourceHint attribute require a valid URI');
+    }
+    path = path.trim();
+    if (path === '') {
+        throw new Error('resourceHint received an empty path');
+    }
+
+    if (typeof state !== 'string') {
+        throw new Error(`resourceHint attribute received invalid value. Valid values are: ${resourceHintStates}`);
+    }
+    state = state.trim();
+    if (!resourceHintStates.includes(state)) {
+        return;
+    }
+
+    if (typeof globals.resourceHints === 'undefined') {
+        globals.resourceHints = [];
```

Comment:
Maybe we create empty array list, when `globals` is initialized? I believe it happens in `index.js`

Reply:
Also, do you think it makes sense to rename this variable to `resourceHintsHeaders`, since it's used only headers?